packages/dataviews/src/dataforms-layouts/panel/index.tsx

```diff
@@ -32,28 +32,43 @@ export default function FormPanelField< Item >( {
 	onChange,
 }: FieldLayoutProps< Item > ) {
 	const { fields } = useContext( DataFormContext );
-	const fieldDefinition = fields.find( ( _field ) => {
-		// Default to the first simple child if it is a combined field.
-		if ( isCombinedField( field ) ) {
-			const simpleChildren = field.children.filter(
-				( child ): child is string | SimpleFormField =>
-					typeof child === 'string' || ! isCombinedField( child )
-			);
-
-			if ( simpleChildren.length === 0 ) {
-				return false;
-			}
-
-			const firstChildFieldId =
-				typeof simpleChildren[ 0 ] === 'string'
-					? simpleChildren[ 0 ]
-					: simpleChildren[ 0 ].id;
-
-			return _field.id === firstChildFieldId;
+	const getSummaryFields = () => {
+		if ( ! isCombinedField( field ) ) {
+			const fieldDef = fields.find( ( _field ) => _field.id === field.id );
+			return fieldDef ? [ fieldDef ] : [];
 		}
 
-		return _field.id === field.id;
-	} );
+		// Use summary field(s) if specified for combined fields
+		if ( field.summary ) {
+			const summaryIds = Array.isArray( field.summary ) 
+				? field.summary 
+				: [ field.summary ];
+			return summaryIds
+				.map( summaryId => fields.find( _field => _field.id === summaryId ) )
+				.filter( Boolean );
```

Comment:
```suggestion
				.filter( field => field !== undefined );
```

[According to TypeScript](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAsiAq5oF4oGdgCcCWA7A5gNwBQxAxgPa4ZT4TABi2ANsBJhACYCSbAtmkSQAygEMAZiigAKAJQAuWAiQBtALpRkAPigBvYlCgdgAV0y4oKqAHIqEawBooJ3Jwji8XJ9eAB3CtZQagB0Hqzs0lDY-Jo60RB8UACEyKgubh64XFCyhFDEAL6klNTAtPRM4Rw8-GgA6tEAFgBCFBTMEKIWqHKKcEIQ6rF6Bkb0ZhZWtlmOzq7unpzefgFBoSxsmJGt7Z0WuflFQA) `filter( Boolean )` doesn't provide a type-safe list (it can return `undefined`, even though it does not).